File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        value_index = {}
        for i, num in enumerate(nums):
            x = target - num
            if x in value_index:
                return [i, value_index[x]]    
            value_index[num] = i
           
           

            
            
        

File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {

    dict := make(map[int]int)
    x := 0
    for i, num := range nums {
        x = target - num
        index, ok := dict[x]
        if ok {
            return []int{index, i}
        }
        dict[num] = i
    } 
    return nil
}